dist/api_data.py

```diff
@@ -557,6 +557,12 @@ def __ge__(self, other):
     Config('debug_mode', '', r'''
         control the settings of various extended debugging features''',
         type='category', subconfig=[
+        Config('background_compact_expire_time', '86400', r'''
+               how long the compact stats for a file should be tracked for before expiring.''',
+               type='int'),
+        Config('background_compact_skip_time', '60', r'''
+               how long a file should sleep before background compact tries again. ''',
```

Comment:
I *think* this is a maximum? I'd call it something more like that: `max_compact_skip_time`, and the description could say: "The maximum amount of time before compact will visit the file" (or similar).

In thinking about these two options, I'm wondering if it makes sense for them to be framed in terms of "passes" or similar. i.e: the maximum number of times compact has visited all tables in the database? Doing that would alleviated some need for exposing these: in a test environment that's likely to be very short. In a production environment with a multi-terabyte database it's likely to be long.

Reply:
Yes, I'm wondering the same. If we do it in terms of passes should it be based on some max attempts made for that particular file or over all files?